get-terraform-provider-versions/README.md

```diff
@@ -0,0 +1,41 @@
+# get-terraform-provider-versions
+
+This folder contains a Python script that will audit Terraform code files.  It will parse these files to look for required_provider statements and use the associated information to build a matrix of used providers.  It will then query the Terraform Registry to find the most recent version available for those providers before finally building output data which helps to identify out-dated and version locked providers.
+
+The default behaviour is to clone all repositories owned by the DFDS organization and perform the analysis on each.  However, some logic is applied as follows:
+
+- Archived repositories will automatically be skipped.
+- Disabled repositories will automatically be skipped.
+- Repositories which do not have HCL listed in their languages section will also be skipped.
+
+## Requirements
+
+A valid GitHub OAuth2 Token needs to be obtained and set using the environment variable named GITHUB_OAUTH2_TOKEN.
+
+### Testing and code coverage
+
+There are currently no test cases, but this is how testing would look like:
+
+```bash
```

Comment:
flake8 will currently fine these issue:

```
./get_terraform_provider_versions/get-tf-provider-versions.py:23:80: E501 line too long (158 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:51:80: E501 line too long (81 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:60:80: E501 line too long (86 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:69:80: E501 line too long (80 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:76:80: E501 line too long (101 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:80:80: E501 line too long (104 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:83:80: E501 line too long (102 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:84:80: E501 line too long (80 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:85:80: E501 line too long (148 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:89:80: E501 line too long (80 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:91:80: E501 line too long (80 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:95:80: E501 line too long (139 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:131:80: E501 line too long (93 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:145:80: E501 line too long (111 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:146:80: E501 line too long (110 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:147:80: E501 line too long (82 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:155:80: E501 line too long (108 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:156:80: E501 line too long (85 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:159:80: E501 line too long (90 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:160:80: E501 line too long (84 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:163:80: E501 line too long (108 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:164:80: E501 line too long (84 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:167:80: E501 line too long (106 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:168:80: E501 line too long (106 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:169:80: E501 line too long (89 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:178:80: E501 line too long (106 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:188:80: E501 line too long (113 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:216:80: E501 line too long (134 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:218:80: E501 line too long (89 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:219:80: E501 line too long (81 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:223:80: E501 line too long (106 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:225:80: E501 line too long (155 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:232:80: E501 line too long (120 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:244:80: E501 line too long (149 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:255:80: E501 line too long (120 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:261:80: E501 line too long (105 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:263:80: E501 line too long (105 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:267:80: E501 line too long (109 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:274:80: E501 line too long (85 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:275:80: E501 line too long (94 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:282:80: E501 line too long (81 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:283:80: E501 line too long (114 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:285:80: E501 line too long (153 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:289:80: E501 line too long (144 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:294:80: E501 line too long (97 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:296:80: E501 line too long (107 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:297:80: E501 line too long (111 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:311:80: E501 line too long (117 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:329:80: E501 line too long (121 > 79 characters)
./get_terraform_provider_versions/get-tf-provider-versions.py:330:80: E501 line too long (127 > 79 characters)
./get_terraform_provider_versions/github/repo.py:15:80: E501 line too long (86 > 79 characters)
./get_terraform_provider_versions/github/repo.py:20:80: E501 line too long (87 > 79 characters)
./get_terraform_provider_versions/github/repo.py:40:80: E501 line too long (81 > 79 characters)
./get_terraform_provider_versions/github/repo.py:45:80: E501 line too long (83 > 79 characters)
./get_terraform_provider_versions/github/repo.py:54:80: E501 line too long (81 > 79 characters)
./get_terraform_provider_versions/github/repo.py:59:80: E501 line too long (83 > 79 characters)
./get_terraform_provider_versions/github/repo.py:69:80: E501 line too long (82 > 79 characters)
./get_terraform_provider_versions/github/repo.py:73:80: E501 line too long (86 > 79 characters)
./get_terraform_provider_versions/github/repo.py:79:80: E501 line too long (87 > 79 characters)
./get_terraform_provider_versions/github/repo.py:84:80: E501 line too long (86 > 79 characters)
./get_terraform_provider_versions/github/repo.py:92:80: E501 line too long (85 > 79 characters)
./get_terraform_provider_versions/github/repo.py:100:80: E501 line too long (84 > 79 characters)
./get_terraform_provider_versions/github/repo.py:103:80: E501 line too long (93 > 79 characters)
./get_terraform_provider_versions/github/repo.py:108:80: E501 line too long (119 > 79 characters)
./get_terraform_provider_versions/github/repo.py:109:80: E501 line too long (88 > 79 characters)
./get_terraform_provider_versions/github/repo.py:155:9: F841 local variable 'toplevel' is assigned to but never used
./get_terraform_provider_versions/github/repo.py:160:9: E265 block comment should start with '# '
./get_terraform_provider_versions/github/repo.py:166:80: E501 line too long (110 > 79 characters)
./get_terraform_provider_versions/github/repo.py:169:16: F821 undefined name 'PROTECTED_REPOS'
./get_terraform_provider_versions/github/repo.py:173:80: E501 line too long (81 > 79 characters)
```

Reply:
In ce-toolbox/get-terraform-provider-versions create a file called .flake8 with the following contents:

```
[flake8]
max-line-length = 88
```

Also run black black get_terraform_provider_versions/

This will reduce the list of linter violations.

Reply:
I've made the config file but then also gone through the Flake8 exceptions and resolved those by adding linebreaks or backslashes where required.